pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -445,3 +476,109 @@ func TestCreateIPSecTunnelPortCert(t *testing.T) {
 		})
 	}
 }
+
+func TestGetNodeMAC(t *testing.T) {
+	mac := "00:1B:44:11:3A:B7"
+	expectedMac, err := net.ParseMAC(mac)
+	require.NoError(t, err)
+	node1 := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{types.NodeMACAddressAnnotationKey: mac},
+		},
+	}
+
+	got, err := getNodeMAC(node1)
+	require.NoError(t, err)
+	assert.Equal(t, expectedMac, got)
```

Comment:
can add cases where the mac is incorrect and cannot be parsed and in the other case you can give empty annotations.